cmake-d/UseDUnittest.cmake

```diff
@@ -7,31 +7,41 @@
 # See LICENSE for details.
 #
 
-macro(add_unittests _sourcefile)
-
-  set(_testname "${_sourcefile}")
-  set(main_unittest "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/unittest.d")
-  set(_source_file "${CMAKE_CURRENT_SOURCE_DIR}/${_sourcefile}")
-
-  #message("main_unittest ${main_unittest}")
-  #message("testname ${_testname}")
-  #message("_sourcefile ${_source_file}")
-
-  if(NOT EXISTS ${main_unittest})
-    file(WRITE ${main_unittest} "// Generated by UseDUnittest.cmake\nint main() { return 0; }")
-  endif()
-
-  # TODO
-  # better way to get the included directories
-  get_directory_property(include_dirs INCLUDE_DIRECTORIES)
-  set(INCLUDES )
-  foreach(include_dir IN LISTS include_dirs)
-    list(APPEND INCLUDES "${CMAKE_INCLUDE_FLAG_D}${include_dir}")
-  endforeach()
-
-  separate_arguments(CMAKE_D_FLAGS)
-
-  add_test(NAME ${_testname}
-  	COMMAND ${CMAKE_D_COMPILER} ${CMAKE_D_FLAGS} ${INCLUDES} ${main_unittest} -unittest -run ${_source_file})
-
+macro(target_add_unittests _target _sourceFileList)
+    set(_testtarget "test_${_target}")
+
+    set(main_unittest "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/unittest.d")
+    if(NOT EXISTS ${main_unittest})
+        file(WRITE ${main_unittest} "// Generated by UseDUnittest.cmake\nint main() { return 0; }")
+    endif()
+   
+    string(REPLACE " " ";" FLAGS ${CMAKE_D_FLAGS_DEBUG_INIT})
```

Comment:
use [separate_arguments](https://cmake.org/cmake/help/v3.0/command/separate_arguments.html)

Reply:
wanted to reuse the debug flags you added in platform files. couldn't pass it as it is because in ctest file it gets then one parameter what dmd doesn't like

Reply:
```
set(UNITTEST_FLAGS ${CMAKE_D_FLAGS_DEBUG_INIT})
separate_arguments(UNITTEST_FLAGS)
```